airflow/contrib/kubernetes/kube_client.py

```diff
@@ -46,6 +46,8 @@ def _load_kube_config(in_cluster, cluster_context, config_file):
 
 
 def get_kube_client(in_cluster=conf.getboolean('kubernetes', 'in_cluster'),
-                    cluster_context=None,
-                    config_file=None):
+                    cluster_context=None if not conf.get('kubernetes', 'cluster_context') else
+                    conf.get('kubernetes', 'cluster_context'),
+                    config_file=None if not conf.get('kubernetes', 'config_file') else
+                    conf.get('kubernetes', 'config_file')):
```

Comment:
These defaults are a bit complex to read as they are. Two things:

1. Leave the default param on the function as None (so have `if cluster_context is None:` in the fn body) - this is due to when Python "evaluates" the default values (at import time) which I would like to not depend on having parsed the config.
2. The condition `if not ... else` can be replaced with `conf.get('kubernetes', 'cluster_context', fallback=None)` if you *don't* specify values in the config. I.e `cluster_context =` becomes `# cluster_context =`

Reply:
Thanks @ashb I have incorporated comments. I have observed that conf.get('kubernetes', 'cluster_context', fallback=None) fallbacks to None if key i.e cluster_context is not defined in airflow.cfg file. In our case it returns a empty string which breaks the kube_client initialisation

Reply:
Which Is why I said

> if you don't specify values in the config. I.e cluster_context = becomes # cluster_context =

That is, make the change in default_airflow.cfg be:


```diff
+# Use the cluster_context or config_file to run outside of cluster
+# cluster_context =
+# config_file =
```